bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -338,8 +348,19 @@ void ReplicaImp::sendAndIncrementMetric(MessageBase *m, NodeIdType id, CounterHa
 void ReplicaImp::onReportAboutInvalidMessage(MessageBase *msg, const char *reason) {
   LOG_WARN(CNSUS, "Received invalid message. " << KVLOG(msg->senderId(), msg->type(), reason));
 
-  // TODO(GG): logic that deals with invalid messages (e.g., a node that sends invalid messages may have a problem (old
-  // version,bug,malicious,...)).
+  // TODO(GG): logic that deals with invalid messages (e.g., a node that sends invalid messages may have a problem
+  // (old version,bug,malicious,...)).
+}
+
+template <>
+void ReplicaImp::onMessage<StateTransferMsg>(StateTransferMsg *m) {
+  if (activeExecutions_ > 0) {
```

Comment:
Is this method happen only on the state transferred replica or on the source also?

Reply:
On both of them.